src/util/irep.cpp

```diff
@@ -537,24 +546,45 @@ int irept::compare(const irept &i) const
   {
     irept::named_subt::const_iterator it1, it2;
 
-    for(it1=get_named_sub().begin(),
-        it2=i.get_named_sub().begin();
-        it1!=get_named_sub().end() && it2!=i.get_named_sub().end();
-        it1++,
-        it2++)
+    for(it1 = get_named_sub().begin(),
+    it2 = i.get_named_sub().begin();
+        it1 != get_named_sub().end() ||
+        it2 != i.get_named_sub().end();) // no iterator increments
     {
+      if(it1 != get_named_sub().end() && is_comment(it1->first))
+      {
+        it1++;
+        continue;
+      }
+
+      if(it2 != i.get_named_sub().end() && is_comment(it2->first))
+      {
+        it2++;
+        continue;
+      }
+
+      // the case that both it1 and it2 are .end() is treated
+      // by the loop guard; furthermore, the number of non-comments
+      // must be the same
+      INVARIANT(it1 != get_named_sub().end(), "not at end of get_named_sub()");
+      INVARIANT(
+        it2 != i.get_named_sub().end(), "not at end of i.get_named_sub()");
```

Comment:
This should basically be the same code as `irept::operator==` but does extra checks and just overall looks a bit different. Maybe we can avoid this?

Reply:
could use compare() for both, but seems to be an unrelated change